go/libraries/doltcore/table/editor/creation/index.go

```diff
@@ -109,6 +110,11 @@ func CreateIndex(
 		return nil, err
 	}
 
+	err = sch.Indexes().SetIndexPrefixLength(indexName, prefixLengths)
```

Comment:
Makes far more sense to have another parameter in `AddIndexByColNames` rather than this new function on the interface